src/main/java/org/folio/inventory/dataimport/handlers/actions/UpdateItemEventHandler.java

```diff
@@ -207,20 +222,81 @@ private void validateStatusName(JsonObject itemAsJson, List<String> errors) {
 
-    itemCollection.update(item).whenComplete((updatedItem, e) -> {
-      if (e != null) {
-        promise.fail(e);
-        return;
-      }
-      promise.complete(updatedItem);
-    });
+    itemCollection.update(item, success -> promise.complete(item),
+      failure -> {
+        if (failure.getStatusCode() == HttpStatus.SC_CONFLICT) {
+          processOLError(item, itemCollection, eventPayload, promise, failure);
+        } else {
+          eventPayload.getContext().remove(CURRENT_RETRY_NUMBER);
+          LOG.error(format("Error updating Item - %s, status code %s", failure.getReason(), failure.getStatusCode()));
+          promise.fail(failure.getReason());
+        }
+      });
     return promise.future();
   }
+
+
+  private void processOLError(Item instance, ItemCollection itemCollection, DataImportEventPayload eventPayload, Promise<Item> promise, Failure failure) {
+    int currentRetryNumber = eventPayload.getContext().get(CURRENT_RETRY_NUMBER) == null ? 0 : Integer.parseInt(eventPayload.getContext().get(CURRENT_RETRY_NUMBER));
+    if (currentRetryNumber < MAX_RETRIES_COUNT) {
+      eventPayload.getContext().put(CURRENT_RETRY_NUMBER, String.valueOf(currentRetryNumber + 1));
+      LOG.warn("OL error updating Instance - {}, status code {}. Retry UpdateItemEventHandler handler...", failure.getReason(), failure.getStatusCode());
+      getActualItemAndReInvokeCurrentHandler(instance, itemCollection, promise, eventPayload);
+    } else {
+      eventPayload.getContext().remove(CURRENT_RETRY_NUMBER);
+      String errMessage = format("Current retry number %s exceeded or equal given number %s for the Item update for jobExecutionId '%s'", MAX_RETRIES_COUNT, currentRetryNumber, eventPayload.getJobExecutionId());
+      LOG.error(errMessage);
+      promise.fail(errMessage);
+    }
+  }
+
+
+  private void getActualItemAndReInvokeCurrentHandler(Item item, ItemCollection itemCollection, Promise<Item> promise, DataImportEventPayload eventPayload) {
+    itemCollection.findById(item.getId())
+      .thenAccept(actualItem -> {
+        JsonObject itemAsJson = convertFieldsInJsonObject(actualItem);
+        eventPayload.getContext().put(ITEM.value(), Json.encode(itemAsJson));
+        eventPayload.getEventsChain().remove(eventPayload.getContext().get(CURRENT_EVENT_TYPE_PROPERTY));
+        try {
+          eventPayload.setCurrentNode(ObjectMapperTool.getMapper().readValue(eventPayload.getContext().get(CURRENT_NODE_PROPERTY), ProfileSnapshotWrapper.class));
+        } catch (JsonProcessingException e) {
+          LOG.error(format("Cannot map from CURRENT_NODE value %s", e.getCause()));
+        }
+        eventPayload.getContext().remove(CURRENT_EVENT_TYPE_PROPERTY);
+        eventPayload.getContext().remove(CURRENT_NODE_PROPERTY);
+        handle(eventPayload).whenComplete((res, e) -> {
+          if (e != null) {
+            promise.fail(e.getMessage());
+          } else {
+            promise.complete();
```

Comment:
Hello @VRohach, please complete the promise by item, otherwise it can cause npe on
dataImportEventPayload.getContext().put(ITEM.value(), ItemUtil.mapToJson(updatedItem).encode());

Reply:
Done. Thank you!